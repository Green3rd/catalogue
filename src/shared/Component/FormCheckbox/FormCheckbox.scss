@import '../../common.scss';

.FormCheckbox {
    font-size: 0.7em;
    display: block;
    position: relative;
    padding-left: 35px;
    margin: 12px 0;
    cursor: pointer;
    user-select: none;

    &__Input {
        position: absolute;
        opacity: 0;
        cursor: pointer;
        height: 0;
        width: 0;
        &:checked {
            & ~ .FormCheckbox__Checkmark {
                background-color: #2196F3;

                &::after {
                    display: block;
                }
            }
        }
    }

    &__Checkmark {
        position: absolute;
        top: 0;
        left: 0;
        height: 25px;
        width: 25px;
        background-color: #eee;
        transition: background-color 0.15s ease-in-out 0s;
        border-radius: 4px;

        &:hover {
            background-color: #ccc;
        }

        &:after {
            content: '';
            position: absolute;
            display: none;
            left: 9px;
            top: 5px;
            width: 5px;
            height: 10px;
            border: solid white;
            border-width: 0 3px 3px 0;
            -webkit-transform: rotate(45deg);
            -ms-transform: rotate(45deg);
            transform: rotate(45deg);
        }
    }

    &__Error {
        color: $errorMessage;
    }
}