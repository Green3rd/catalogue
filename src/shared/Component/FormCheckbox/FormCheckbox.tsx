import React from "react";
import "./FormCheckbox.scss";
import { ErrorMessage, Field } from "formik";

interface Props {
  label: string;
  name: string;
}

const FormCheckbox = (props: Props) => {
  const { label, name } = props;
  return (
    <label className="FormCheckbox">
      <Field className="FormCheckbox__Input" type="checkbox" name={name} />
      <span className="FormCheckbox__Checkmark" />
      {label}
      <ErrorMessage
        className="FormCheckbox__Error"
        name={props.name}
        component="div"
      />
    </label>
  );
};
FormCheckbox.displayName = "FormCheckbox";
export { FormCheckbox };
