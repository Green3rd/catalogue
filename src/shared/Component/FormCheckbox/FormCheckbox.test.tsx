import { shallow } from 'enzyme';
import { FormCheckbox } from './FormCheckbox';
import enzymeToJson from 'enzyme-to-json';

const setup = () => {
  const ownProps = {
    label: 'text',
    name: 'errorText',
  };

  const props = {
    ...ownProps,
  };

  const Component = shallow(<FormCheckbox {...props} />);
  return { Component, props };
};

describe('<FormCheckbox />', () => {
  it('should render and function correctly', () => {
    const { Component } = setup();
    expect(enzymeToJson(Component)).toMatchSnapshot();
  });
});
