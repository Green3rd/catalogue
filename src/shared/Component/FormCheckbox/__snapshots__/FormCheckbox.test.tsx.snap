// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormCheckbox /> should render and function correctly 1`] = `
<label
  className="FormCheckbox"
>
  <Field
    className="FormCheckbox__Input"
    name="errorText"
    type="checkbox"
  />
  <span
    className="FormCheckbox__Checkmark"
  />
  text
  <FormikConnect(ErrorMessageImpl)
    className="FormCheckbox__Error"
    component="div"
    name="errorText"
  />
</label>
`;
